client.go

```diff
@@ -1830,3 +1836,41 @@ func flags(f int) uint32 {
 	}
 	return out
 }
+
+// unixPerm converts Go permission bits to Unix permission bits,
+// masking off file type bits.
+//
+// The SFTP protocol draft doesn't specify the exact values of the bits.
+// POSIX is fuzzy in this regard: it specifies bits for the chmod command [1],
+// but not for C applications [2]. We follow the chmod bits, which are the
```

Comment:
I feel like the [2] reference here is essentially irrelevant, the internal details of how the C structures store data is immaterial to the values accepted and used by `chmod`. Any POSIX implementation will need to convert these strictly defined chmod values to any internal implementation regardless of the internal representation being unspecified.

Perhaps naming this `toChownPerms(…)` might be the best choice? This should make it clear that we are converting to chmod-specific permissions, rather than just a nebulous “POSIX permissions”.

Reply:
The thing is that OpenSSH's sftp-server sends the actual C bits in response to a stat request: https://github.com/openssh/openssh-portable/blob/82fef71e20ffef425b932bec26f5bc46aa1ed41c/sftp-common.c#L78

Whether this is intentional is not explained in the 2001 protocol draft.

Reply:
The [old SFTP draft](https://tools.ietf.org/html/draft-ietf-secsh-filexfer-02#section-5) refers explicitly to the POSIX C API, without noticing that it doesn't specify the bit values. Or maybe it in the 1996 edition...

In any case, if you say "stop whining and use sycall", I'll comply. I'll rename the function toChownPerms in any case.